pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java

```diff
@@ -134,12 +134,15 @@ public void run() throws Exception {
     Path outputDir = new Path(_outputDir);
 
     if (fs.exists(stagingDir)) {
-      LOGGER.warn("Found the temp folder, deleting it");
+      LOGGER.warn("Found the temp folder {}, deleting it", stagingDir);
       fs.delete(stagingDir, true);
     }
     fs.mkdirs(stagingDir);
-    if (_defaultPermissions != null) {
-      FileSystem.get(getConf()).setPermission(stagingDir, new FsPermission(_defaultPermissions));
+
+    if (_defaultPermissionsMask != null) {
+      FsPermission umask = new FsPermission(_defaultPermissionsMask);
+      FsPermission permission = FsPermission.getDefault().applyUMask(umask);
```

Comment:
What does FsPermission.getDefault() return? Is it guaranteed to be 777? If not, this will not work.

Also, lets log the permissions being set.